DotNetHowToSynonyms/DotNetHowToSynonyms/Program.cs

```diff
@@ -86,13 +89,31 @@ private static void CreateHotelsIndex(SearchServiceClient serviceClient)
             serviceClient.Indexes.Create(definition);
         }
 
-        private static void EnableSynonymsInHotelsIndex(SearchServiceClient serviceClient)
+        private static void EnableSynonymsInHotelsIndexSafely(SearchServiceClient serviceClient)
         {
             Index index = serviceClient.Indexes.Get("hotels");
+            index = AddSynonymMapsToFields(index);
+
+            try
+            {
+                // The IfMatchCondition ensure that the index is updated only if the ETags match.
+                serviceClient.Indexes.CreateOrUpdate(index, accessCondition: AccessCondition.GenerateIfMatchCondition(index.ETag));
```

Comment:
There's a shorthand for this. You can just call `AccessCondition.IfNotChanged(index)` instead.